File: clients/elixir_client/benchmark.exs
defmodule SoliDB.Benchmark do
  def run do
    {:ok, client} = SoliDB.Client.connect()
    :ok = SoliDB.Client.auth(client, "_system", "admin", "password")

    db = "bench_db"
    col = "elixir_bench"

    SoliDB.Client.query(client, "_system", "CREATE DATABASE #{db}")
    SoliDB.Client.query(client, db, "CREATE COLLECTION #{col}")

    iterations = 1000
    
    start_time = System.monotonic_time()
    
    for i <- 1..iterations do
      SoliDB.Client.insert(client, db, col, %{id: i, data: "benchmark data content"})
    end
    
    end_time = System.monotonic_time()
    duration = System.convert_time_unit(end_time - start_time, :native, :microsecond) / 1_000_000
    
    ops_per_sec = iterations / duration
    
    IO.puts("ELIXIR_BENCH_RESULT:#{:erlang.float_to_binary(ops_per_sec, [decimals: 2])}")
  end
end

SoliDB.Benchmark.run()

File: clients/go-client/benchmark.go
package main

import (
	"fmt"
	"time"

	"github.com/solisoft/solidb-go-client/solidb"
)

func main() {
	client := solidb.NewClient("127.0.0.1", 9999)
	err := client.Connect()
	if err != nil {
		panic(err)
	}
	defer client.Close()

	client.Auth("_system", "admin", "admin")

	db := "bench_db"
	col := "go_bench"

	// Ignore errors if already exists
	client.CreateDatabase(db)
	client.CreateCollection(db, col, nil)

	iterations := 1000

	startTime := time.Now()
	for i := 0; i < iterations; i++ {
		data := map[string]interface{}{
			"id":   i,
			"data": "benchmark data content",
		}
		_, err = client.Insert(db, col, data, nil)
		if err != nil {
			fmt.Printf("Error: %v\n", err)
		}
	}
	duration := time.Since(startTime)

	opsPerSec := float64(iterations) / duration.Seconds()

	fmt.Printf("GO_BENCH_RESULT:%.2f\n", opsPerSec)
}
